addons/stock/models/stock_move.py

```diff
@@ -416,51 +416,86 @@ def onchange_date(self):
         if self.date_expected:
             self.date = self.date_expected
 
-    # TDE DECORATOR: remove that api.multi when action_confirm is migrated
-    @api.multi
+    def _key_assign_picking(self):
+        self.ensure_one()
+        return self.group_id, self.location_id, self.location_dest_id, self.picking_type_id
+
+    def _search_picking_for_assignation(self):
+        self.ensure_one()
+        picking = self.env['stock.picking'].search([
+                ('group_id', '=', self.group_id.id),
+                ('location_id', '=', self.location_id.id),
+                ('location_dest_id', '=', self.location_dest_id.id),
+                ('picking_type_id', '=', self.picking_type_id.id),
+                ('printed', '=', False),
+                ('state', 'in', ['draft', 'confirmed', 'waiting', 'partially_available', 'assigned'])], limit=1)
+        return picking
+
     def assign_picking(self):
         """ Try to assign the moves to an existing picking that has not been
         reserved yet and has the same procurement group, locations and picking
         type (moves should already have them identical). Otherwise, create a new
         picking to assign them to. """
         Picking = self.env['stock.picking']
+        grouped_moves = {}
```

Comment:
On pourrait rajouter les commentaires # OF Modification OpenFire et Fin autour de l'ensemble de ces modifs ?